### PR TITLE
Create and fetch chart data and display timeline chart on portfolio 

### PR DESCRIPTION
**Pull request by dhruvi618**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java [hunk 1]
```diff
@@ -45,22 +47,14 @@ public void init() {
...
-      int endMonth = Integer.parseInt(endDateComponents[0]);
-      int endDay = Integer.parseInt(endDateComponents[1]);
-      int endYear = Integer.parseInt(endDateComponents[2]);
-      Date endDate = new Date(endYear - 1900, endMonth - 1, endDay);
+      // Parse dates using mm/dd/yyyy pattern and create corresponding Date objects
+      SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+      try {  
+        Date startDate = formatter.parse(startDateString);
+        Date endDate = formatter.parse(endDateString);
+      } catch (ParseException e) {e.printStackTrace();}
```
[the remark this answers is not in the document]
Comment (dhruvi618) [reply portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java]: I decided to go with sending back a status 500 and outputting a user-readable error message to the screen.

File: portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java [hunk 2]
```diff
@@ -50,20 +51,30 @@ public void init() {
 
       // Parse dates using mm/dd/yyyy pattern and create corresponding Date objects
       SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+      formatter.setLenient(false);
```
Comment (kmh287) [inline portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java]: The formatter can be defined outside of the loop

File: portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java [hunk 3]
```diff
@@ -34,6 +34,7 @@
 public class ProgrammingExperienceServlet extends HttpServlet {
 
   private List<ProgrammingExperience> programmingExperiences = new ArrayList<>();
+  boolean parseError = false;
```
Comment (kmh287) [inline portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java:37]: How about something more specific like `programmingLanguageExperienceParseError`?

File: portfolio/src/main/webapp/WEB-INF/programming-language-experience.csv [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Java,09/01/2015,06/01/2020
+C,09/01/2019,06/01/2020
+Javascript,05/01/2020,06/01/2020
+OCaml,01/01/2020,06/01/2020
```
[the remark this answers is not in the document]
Comment (kmh287) [reply portfolio/src/main/webapp/WEB-INF/programming-language-experience.csv:4]: My school had the same class. To be honest I can't remember anything about OCaml anymore. 😆

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -56,50 +56,44 @@ function pauseProjectChange() {
...
-      textStyle: {
-        color: '#ffffff'
-      }
-    },
-    titleTextStyle: {
-      color: '#ffffff'
+  fetch('/programming-data').then(function(response) {
+    const chartContainer = document.getElementById('chart-container');
+    if (response.status === 500) {
+      chartContainer.appendChild(createParagraphElement("Error parsing chart data"));
```
Comment (kmh287) [inline portfolio/src/main/webapp/script.js:70]: The date parsing error is only one of a variety of reasons why the server could return a 500. Could the server include the error message in its response? Then the client could use that message when provided and use something more generic like "Error fetching chart data" when not provided.

File: portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java [hunk 6]
```diff
@@ -69,8 +70,8 @@ public void init() {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Send error code 500 in event of parsing exception and programming experiences JSON on success
-    if (parseError) {
-      response.setStatus(500);
+    if (programmingLanguageExperienceParseError) {
+      response.sendError(500,"Error parsing chart data");
```
Comment (kmh287) [inline portfolio/src/main/java/com/google/sps/servlets/ProgrammingExperienceServlet.java]: space after the comma.

**Pull request merged**